src/v2/components/ChannelContents/index.tsx

```diff
@@ -41,244 +47,243 @@ interface ChannelContentsProps extends Props {
   client: ApolloClient<any>
 }
 
-const ChannelContents: React.FC<ChannelContentsProps> = ({
-  chunkSize = 10,
-  channel,
-  client,
-  moveConnectable,
-  ...rest
-}) => {
-  // Used to load/unload waypoints
-  const [activeQueries, setActiveQueries] = useState<
-    ActiveQueriesCollection.ActiveQueries
-  >({})
-
-  // Handles ordering of block grid items
-  const [connectables, setConnectables] = useState(channel.skeleton)
-
-  // Handles actual contents of block grid items
-  const [collection, setCollection] = useState(
-    ConnectableCellsCollection.normalize(channel.initial_contents)
-  )
-
-  const addConnectable = useCallback(newConnectable => {
-    setConnectables(prevConnectables => {
-      if (
-        !prevConnectables.some(
-          c =>
-            c.id === newConnectable.id &&
-            c.type.toUpperCase() === newConnectable.type.toUpperCase()
-        )
-      ) {
-        return [
-          {
-            id: newConnectable.id,
-            type: { BLOCK: 'Block', CHANNEL: 'Channel' }[
-              newConnectable.type.toUpperCase()
-            ],
-            __typename: 'SkeletalConnectable',
-          },
-          ...prevConnectables,
-        ]
-      }
-
-      return prevConnectables
-    })
-  }, [])
-
-  const updateConnectable = useCallback(
-    connectable => {
-      loadSkeleton({
-        client,
-        channelId: channel.id,
-        pageSkeleton: [connectable],
-        collection,
-        queryOptions: {
-          fetchPolicy: 'network-only',
-        },
-      }).then(contents => {
-        if (!contents) return
-        setCollection(prevCollection => ({ ...prevCollection, ...contents }))
-      })
-    },
-    [channel.id, client, collection]
-  )
-
-  const parsePayload = useCallback(
-    ({ id, base_class }) => ({
-      id,
-      type: base_class.toUpperCase(),
-    }),
-    []
-  )
-
-  usePusher({
-    socketId: `channel-${NODE_ENV}-${channel.id}`,
-    onCreated: addConnectable,
-    onUpdated: updateConnectable,
-    parsePayload,
-  })
-
-  const chunked = useMemo(() => chunk(connectables, chunkSize), [
-    connectables,
-    chunkSize,
-  ])
-
-  const handleAddBlock = useCallback(
-    ({ id }: { id: number }) => {
-      addConnectable({ id, type: 'Block' })
-    },
-    [addConnectable]
-  )
-
-  const handleRemoveBlock = useCallback(
-    ({ id, type }: { id: number; type: string }) => {
+const ChannelContents: React.FC<ChannelContentsProps> = memo(
+  ({ chunkSize = 10, channel, client, moveConnectable, ...rest }) => {
+    // Used to load/unload waypoints
+    const [activeQueries, setActiveQueries] = useState<
+      ActiveQueriesCollection.ActiveQueries
+    >({})
+
+    // Handles ordering of block grid items
+    const skeleton: ChannelContentsSkeletonData[] = isClientSide
+      ? channel.skeleton
+      : first(channel.skeleton, chunkSize)
+    const [connectables, setConnectables] = useState(skeleton)
```

Comment:
This appears to work pretty seamlessly locally. Will see if it has any real effect 🤞